peerdid/peer_did.py

```diff
@@ -1,68 +1,113 @@
+import json
 import re
 from typing import List, Optional
 
-from peerdid.storage import Storage, FileStorage
+from peerdid.peer_did_utils import _encode_service, _create_encnumbasis, \
+    _build_did_doc_numalgo_0, _build_did_doc_numalgo_2, _check_key_correctly_encoded, _is_json
 from peerdid.types import PEER_DID, PublicKeyAgreement, PublicKeyAuthentication, JSON
 
 
 def is_peer_did(peer_did: PEER_DID) -> bool:
     """
     Checks if peer_did parameter matches Peer DID spec
     (https://identity.foundation/peer-did-method-spec/index.html#matching-regex)
+
     :param peer_did: peer_did to check
-    :return: true if peer_did matches spec, otherwise false
+
+    :return: True if peer_did matches spec, otherwise False
     """
     if peer_did is None:
         return False
-    peer_did_pattern = re.compile(r'^did:peer:[01](z)([1-9a-km-zA-HJ-NP-Z]{46,47})$')
+    peer_did_pattern = re.compile(
+        r"^did:peer:(([0](z)([1-9a-km-zA-HJ-NP-Z]{46,47}))|(2((\.[AEVID](z)([1-9a-km-zA-HJ-NP-Z]{46,47}))+"
```

Comment:
sorry if I missed this discussion but is there a reason to change from the regex listed [in the spec](https://identity.foundation/peer-did-method-spec/#matching-regex)?

Reply:
The regexp in the spec appeared to be incorrect (not full) as it doesn't support numalgo2 methods for peer did creation.